prime-router/src/main/kotlin/history/SubmissionHistory.kt

```diff
@@ -512,15 +512,13 @@ class DetailedSubmissionHistory(
         }
 
         if (destinations.size == 0) {
-            /**
-             * Cases where this may hit:
-             *     1) Data hasn't been processed yet (common in async submissions)
-             *     2) Very rare: No data matches any geographical location.
-             *         e.g. If both the testing tab and patient data were foreign addresses.
-             * At the moment we have NO easy way to distinguish the latter rare case,
-             * so it will be treated as status RECEIVED as well.
-             */
-            return Status.RECEIVED
+            return if (reportItemCount != 0) {
+                // Data hasn't been processed yet (common in async submissions)
+                Status.RECEIVED
+            } else {
+                // Very rare: No data matches any geographical location.
```

Comment:
Is this comment accurate? A report can end up passing the jurisdictional filter (geographical location) but fail another one, like the quality filter, and go nowhere, right?

Reply:
I can look into this some more, but I was just maintaining the existing comment.

Reply:
So I think the under case is handled by the conditional starting on 522L.  `realDestinations` are the original destinations but filtered down to the ones that they actually had an item delivered.